# HG changeset patch
# User Andreas Weber <octave@josoansi.de>
# Date 1546465565 -3600
#      Wed Jan 02 22:46:05 2019 +0100
# Branch stable
# Node ID 3f367754063469daaa9a7c852a5af89ecd587609
# Parent  00f796120a6d5c25897fd9794feed868938f15dd
file-io.cc: Fix static analyzer detected V595 issue (bug #55347)

diff --git a/libinterp/corefcn/file-io.cc b/libinterp/corefcn/file-io.cc
--- a/libinterp/corefcn/file-io.cc
+++ b/libinterp/corefcn/file-io.cc
@@ -428,23 +428,25 @@ do_stream_open (const std::string& name,
 
   if (! fs.is_dir ())
     {
 #if defined (HAVE_ZLIB)
       if (use_zlib)
         {
           FILE *fptr = std::fopen (fname.c_str (), mode.c_str ());
 
-          int fd = fileno (fptr);
-
-          gzFile gzf = ::gzdopen (fd, mode.c_str ());
-
           if (fptr)
-            retval = octave_zstdiostream::create (fname, gzf, fd,
-                                                  md, flt_fmt);
+            {
+              int fd = fileno (fptr);
+
+              gzFile gzf = ::gzdopen (fd, mode.c_str ());
+
+              retval = octave_zstdiostream::create (fname, gzf, fd,
+                                                    md, flt_fmt);
+            }
           else
             retval.error (std::strerror (errno));
         }
       else
 #endif
         {
           FILE *fptr = std::fopen (fname.c_str (), mode.c_str ());
 
